custom_components/upnp_availability/upnpstatustracker.py

```diff
@@ -194,16 +195,26 @@ async def find_devices(self, async_callback=None):
             async_callback = self.handle_alive
 
         for source_address in self.source_addresses:
+            _LOGGER.debug("Starting device search using addr %s", source_address)
             try:
-                await async_search(
-                    service_type=ROOT_DEVICE,
-                    source_ip=ip_address(source_address),
-                    async_callback=async_callback,
-                )
-            except OSError as ex:
-                _LOGGER.warning(
-                    "Unable to search using addr %s: %s", source_address, ex
-                )
+                if type(ip_address(source_address)) is IPv6Address:
```

Comment:
Use `ip_address` to create the instance here, and reuse it for the later checks:
```
try:
    addr = ip_address(source_address)
except ValueError:
    logging.warning("Unable to search ..")
    continue
```

Also, instead of checking for the type, use `version` property to check for the version, and prepare the address to be passed to a new internal search function:
```
if addr.version == 6:
    # prepare the address for v6 search here
else:
    # prepare the address for v4 sesarch

self._search(addr)
```